nkms/crypto/powers.py

```diff
@@ -21,80 +22,62 @@ class NoEncryptingPower(PowerUpError):
 
     def consume_power_up(self, power_up):
         if isinstance(power_up, CryptoPowerUp):
             power_up_class = power_up.__class__
             power_up_instance = power_up
         elif CryptoPowerUp in inspect.getmro(power_up):
             power_up_class = power_up
-            power_up_instance = power_up(generate_keys_if_needed=self.generate_keys)
+            power_up_instance = power_up(
+                generate_keys_if_needed=self.generate_keys)
         else:
             raise TypeError(
                 ("power_up must be a subclass of CryptoPowerUp or an instance "
                  "of a subclass of CryptoPowerUp."))
         self._power_ups[power_up_class] = power_up_instance
 
         if power_up.confers_public_key:
-            # TODO: Make this an ID for later lookup on a KeyStore.
-            self.public_keys[
-                power_up_class] = power_up_instance.public_key()
+            self.public_keys[power_up_class] = power_up_instance.public_key()
 
     def pubkey_sig_bytes(self):
         try:
-            # TODO: Turn this into an ID lookup on a KeyStore.
-            return self._power_ups[
-                SigningPower].pub_key
-        except KeyError:
-            raise NoSigningPower
-
-    def pubkey_sig_tuple(self):
-        try:
-            # TODO: Turn this into an ID lookup on a KeyStore.
-            return API.ecdsa_bytes2pub(self._power_ups[
-                                           SigningPower].pub_key)
+            pubkey_sig = self._power_ups[SigningPower].public_key()
+            return bytes(pubkey_sig)
         except KeyError:
             raise NoSigningPower
 
-    def sign(self, *messages):
+    def sign(self, message):
         """
-        Signs a message and returns a signature with the keccak hash.
-        :param Iterable messages: Messages to sign in an iterable of bytes
-        :rtype: bytestring
-        :return: Signature of message
+        TODO: New docstring.
         """
         try:
             sig_keypair = self._power_ups[SigningPower]
         except KeyError as e:
             raise NoSigningPower(e)
-        msg_digest = b"".join(API.keccak_digest(m) for m in messages)
-
-        return Signature(sig_keypair.sign(msg_digest))
+        return sig_keypair.sign(message)
 
-    def decrypt(self, ciphertext):
+    def decrypt(self, message_kit):
         try:
             encrypting_power = self._power_ups[EncryptingPower]
-            return encrypting_power.decrypt(ciphertext)
+            return encrypting_power.decrypt(message_kit)
         except KeyError:
             raise NoEncryptingPower
 
-    def encrypt_for(self, pubkey, cleartext):
-        try:
-            encrypting_power = self._power_ups[EncryptingPower]
-            ciphertext = encrypting_power.encrypt(cleartext, bytes(pubkey))
-            return ciphertext
-        except KeyError:
-            raise NoEncryptingPower
+    def encrypt_for(self, recipient_pubkey_enc, plaintext):
```

Comment:
Love these red lines, this file looks so much better now! Great job, everyone!

Reply:
✔️